### ISSUE TITLE
Ability to update prefills with visual selection 

### ISSUE THREAD

**Issue by mehalter** [MagicDuck/grug-far.nvim#258]
I love the improvements to the API and particularly enjoy the ability to update the prefills of a given instance. Given that the prefilling with visual selection is a separate command, could there be an addition to the API to allow updating the prefills with the current visual selection as well?

**Comment by MagicDuck** [comment]
hi @mehalter,

So the way visual selection currently works is a bit special in that depending on the engine (ripgrep/astgrep) we put in additional flags. For example, for ripgrep, we add `--multiline --fixed-strings`. This works on initial launch but would feel weird on an `update_instance_prefills()` call.

I think what we can do though and would be the most flexible is expose a utility function to get the current visual selection as  a string, that users can then pass to `update_instance_prefills()`. Seems like a common enough thing and would be annoying to rewrite.  

Implementation would look something like:
```lua
function M.get_current_visual_selection()
  local isVisualMode = vim.fn.mode():lower():find('v') ~= nil
  if isVisualMode then
    -- needed to make visual selection work
    vim.cmd([[normal! vv]])
  end

  local selection_lines = utils.getVisualSelectionLines()
  return vim.fn.join(selection_lines, '\n')
end
```

So you would call it like:
```lua
grugFar.update_instance_prefills('myinstance', { search = grugFar.get_current_visual_selection() })
```
The beauty of this is that you can modify it before passing it in or prefill some other input like replace.

**Comment by mehalter** [comment]
Oh yeah that looks great!

**Comment by MagicDuck** [comment]
let me know if you see any issue, it was sort of hastily thrown in 😄

**Comment by mehalter** [comment]
I'll let you know tomorrow when I mess around with it. Only thing that comes immediatly to mind is it seems like there might be a good way to interpolate this through the `open` and `with_visual_selection` functions to save on duplicate code

**Comment by MagicDuck** [comment]
Those ones already use the util function, and are slightly different since they need to call on engine specific logic, but if there is a nice way to do it I am not against it…

**Comment by mehalter** [comment]
@MagicDuck I took a stab at this while also fixing a few bugs in the visual selection  code. Mainly bugs in how the code was leaving visual mode which was exposed when using the `update_prefills` method. And another bug that was breaking visual line mode. I opened up a PR that should be good for review now. This refactor did lead to a nice way to centralize the semantics of "getting a visual selection" so that it is retrieved the same way across the code base. Hope this helps! I'm glad to get to contribute to a great project :D

PR #265
